bsip-00XX Sharedrop Operation.md

```diff
@@ -0,0 +1,153 @@
+    BSIP: TBD
+    Title: Sharedrop operation
+    Authors: OpenLedgerApp <https://github.com/OpenLedgerApp>
+    Status: Draft
+    Type: Protocol
+    Created: 2018-12-17
+    Updated: 2018-12-17
+    Discussion: 
+    Worker: TBD
+ 
+A sharedrop is a giveaway of some tokens to a certain group of people, for example to all BTS holders in proportion to their holdings.
+
+# Abstract
+We should add the ability to distribute arbitrary amount of some token (SHAREDROP token) to holders of a specific other token (STAKE token)  according to the share(stake) of such tokens.
+The importance of a similar solution was raised in BSIPs-19, 20,  however never implemented. (link below).
+
+
+# Motivation
+By implementing this proposal, we let Asset Owners or other businesses an opportunity to distribute their profit to the holders of a specific asset in proportion of the owned amount of this asset.
+
+Also, usual token drops can be done via the sharedrop operation.
+
+# Rationale
+
+- Other cryptocurrency platforms offer profit-sharing/dividends functionality, such as Peerplays/NXT/CounterParty/DigixDAO/LBRY/Waves/Dash.
+- Most (if not all) Bitcoin forks (1000's of them) have 'sendmany' functionality which enables the mass distribution of their tokens against a large quantity of users (addresses) in a single transaction; this is currently not possible within BitShares.
+
+To make it more difficult to abuse the system dropping multiple useless tokens, there will be a vesting stake for the sharedrop operation. For example, vest 5000 bitUSD (in BTS) for 2 month to be able to do sharedrop operation. 
+
+Sharedrop operation should also be quite expensive as well. 
+However, to make such an operation more lucrative for BtiShares - the sharedrop operation will send the money to each stakeholder's vesting balance. Thus making them withdraw the drop from the vesting balances.
+
+ 
+# Specifications
+## General workflow
+First of all, there is a need to calculate transaction fee, so **dry_run_sharedrop** operation should be called on a local node to do calculations.  THe results will be provided (displayed) to a user. 
+
+Then, knowing that user has enough BTS for transaction fees and vesting money, he should call Sharedrop operation itself. Vesting specifics will be covered below. 
+
+Sharedrop operation will drop the Sharedrop asset in proportion to the stake that stakeholders have to their respective vesting balances.
+
+## dry_run_sharedrop operation 
```

Comment:
Please see above. If we change the fee to the FIXED fee then we do not need it at all. 
if we still have the fee depend on the number of users, then we still need to calculate it. 
In this case, we can definitely extend the ``` get_required_fees ``` for this operation and implement our calculations there.
Thank you for the idea.
Again, it depends on the discussion about the fees and vesting stake.

Reply:
What is the idea which vesting balance object to use?

If the user does not have a vesting balance object, it will need to be created inducing, i.e. more fees. IMO it can't be a fixed fee in such a case